pkg/virt-controller/watch/vm.go

```diff
@@ -1295,6 +1295,10 @@ func (c *VMController) startVMI(vm *virtv1.VirtualMachine) (*virtv1.VirtualMachi
 	log.Log.Object(vm).Infof("Started VM by creating the new virtual machine instance %s", vmi.Name)
 	c.recorder.Eventf(vm, k8score.EventTypeNormal, SuccessfulCreateVirtualMachineReason, "Started the virtual machine by creating the new virtual machine instance %v", vmi.ObjectMeta.Name)
 
+	if hasStartRequest(vm) {
```

Comment:
Why is this required?

Reply:
Bunch of reasons, first off is simplification, why delaying something if you already know it can be done? 

Second, if you remove those pops() in Start()/Stop() the newly added unit test does not pass as `isTrimFirstChangeRequestNeeded()` only trims a Stop request if `vmi == nil` which is never the case if the VMI gets immediately deleted by the API server (like in unit tests), hence I smell somewhat a racy behaviour here. Basically the stop requests never get trimmed if deletion is very quick.

Instead of fixing the `isTrimFirstChangeRequestNeeded` I opted for this because I think it's just easier and more logical to act immediately, what I'd like to do next in a separate PR is removing that trim logic entirely.